tools/acorn-optimizer.js

```diff
@@ -2,7 +2,7 @@
 
 'use strict';
 
-const acorn = require('acorn');
```

Comment:
I guess that idea here is that you want to remove the package.json file from your distribution of emscripten?   
Could you avoid this change by instead just emptying it out?  i.e. remove all the deps from it, but have it still be there to act as root so that that our `node_modules` can be located?

Perhaps we should do the same thing for emsdk, so that we know we have good test coverage?

Reply:
Yeah, I want to delete the package.json file in our Unity redistribution, at the root, or possibly even recursively in all subdirectories. Looks like the CI run came back green for when one would just delete the root package.json files, so that seems functionally safe thing to do, if we only change this one line.

I would prefer to change the acorn lookup here to be manual so we don't need to rely on package.json at all, since otherwise I would need to maintain stub package.jsons in our redistributable creator script. Manual lookup here does not look so bad, as terser already is being manually found as well anyways? (on the next line)

I am not proposing that package.jsons would (necessarily) be deleted from the emscripten installations that google's waterfall does, but that the test runs that we are running on Emscripten CircleCI would always run in this mode where these package.json files (either at root, or recursively) are deleted, so that we know that running in this mode will remain safe thing for us to rely on. (I don't anticipate that we would evolve any features that would actually need these)

> Perhaps we should do the same thing for emsdk, so that we know we have good test coverage?

Hmm, emsdk does not use node_modules/package.json, or maybe I misunderstood here?

Reply:
I think both unity's distribution and emsdk's distribution expect `node_modules` to be pre-existing and don't expect to ever run `npm` to add/remove/update stuff.

So in both/all cases I think we can strip the `package.json` file when we ship. 

I would rather ship an empty one so that we can avoid absolute paths..  but maybe there is some other way do that.

Reply:
Also this whole thing strikes me as rather crazy.. since we not actually removing any of the insecure code, just making a little harder for the security scanners to find it :-/    I'm not sure that's a good direction to go in.